sunpy/map/compositemap.py

```diff
@@ -444,7 +448,29 @@ def plot(self, axes=None, annotate=True,
             if m.levels is False:
                 # We tell GenericMap.plot() that we need to autoalign the map
                 if wcsaxes_compat.is_wcsaxes(axes):
-                    params['autoalign'] = True
+                    # Set 'autoalign' to True if `m.wcs` differs from `axes.wcs`
+                    # otherwise, False.
+                    params['autoalign'] = False
+                    for attr in COMPARE_WCS_ATTR:
+                        m_attr = getattr(m.wcs.wcs, attr, None)
+                        axes_attr = getattr(axes.wcs.wcs, attr, None)
+                        if np.all(np.array(m_attr) != np.array(axes_attr)):
+                            params['autoalign'] = True
+
+                    if not params['autoalign']:
+                        m_rsun_ref = getattr(m.wcs.wcs.aux, 'rsun_ref', None)
+                        axes_rsun_ref = getattr(axes.wcs.wcs.aux, 'rsun_ref', None)
+                        if m_rsun_ref != axes_rsun_ref:
+                            params['autoalign'] = True
+
+                    if not params['autoalign']:
+                        m_pc = getattr(m.wcs.wcs, 'pc', None)
+                        axes_pc = getattr(axes.wcs.wcs, 'pc', None)
+                        m_crota = getattr(m.wcs.wcs, 'crota', None)
+                        axes_crota = getattr(axes.wcs.wcs, 'crota', None)
+
+                        if np.all(np.array(m_pc) != np.array(axes_pc)) or (m_crota != axes_crota):
+                            params['autoalign'] = True
```

Comment:
```suggestion
                    # Set 'autoalign' to True if `m.wcs` differs from `axes.wcs`
                    params['autoalign'] = not axes.wcs.wcs.compare(m.wcs.wcs, tolerance=0.01)
```
If `compare()` is good enough to use [in `GenericMap`](https://github.com/sunpy/sunpy/blob/5668bba8ae896c8147c8db4fc7be88b2b04b1fa9/sunpy/map/mapbase.py#L2923), we should do the same in `CompositeMap`.

Reply:
```
import sunpy.data.sample
import sunpy.map

aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
hmi_map = sunpy.map.Map(sunpy.data.sample.HMI_LOS_IMAGE)
out_map = hmi_map.reproject_to(aia_map.wcs)

out_map.wcs.wcs.compare(aia_map.wcs.wcs)  # // False
out_map.wcs.wcs.compare(m.wcs.wcs, tolerance=0.01)  # True
```

earlier i have used `wcs.wcs.compare()` this returns  False 
Thats'why i implemented this code